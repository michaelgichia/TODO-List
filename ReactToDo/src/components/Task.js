import React from 'react';

export default (props) => {
	return (
		<div className="task">
		  <label className="waves-effect waves-light" id="all-todo" onClick={props.displayAllTodos}>All</label>
		  <label className="waves-effect waves-light" id="toggle-all" onClick={props.toggleAll}>Toggle</label>
		  <label className="waves-effect waves-light" id="active-todo" onClick={props.displayActiveTodos}>Active</label>
		  <label className="waves-effect waves-light" id="complete-todo" onClick={props.displayCompletedTodos}>Completed</label>
		</div>
	);
}
