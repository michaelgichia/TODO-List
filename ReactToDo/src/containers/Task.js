import React, { Component, PropTypes } from 'react';
import _ from 'lodash';
import Label from '../components/Label';

export default class Task extends Component {

  toggleAll = () => {
    const {todos} = this.props;
    let activeTodos = this.activeTodos(todos);
    let completedTodos = this.completedTodos(todos);
    const newTodos = this.newTodos(todos, activeTodos, completedTodos)
    this.props.updateState(newTodos)
  }

  activeTodos = (todos) => {
    return todos.filter(todo => !todo.isCompleted);
  }

  completedTodos = (todos) => {
    return todos.filter(todo => !!todo.isCompleted);
  }

  newTodos = (todos, activeTodos, completedTodos) => {
    if(completedTodos.length > 0) {
      return todos.map(todo => Object.assign({}, todo, {isCompleted: false}))
    } else if(activeTodos.length > 0) {
      return todos.map(todo => Object.assign({}, todo, {isCompleted: true}))
    } else {
      return todos.map(todo => Object.assign({}, todo, {isCompleted: !todo.isCompleted}))
    }
  }

	render () {
		return (
			<div className="task">
			  <Label id="all-todo" onClick={this.props.displayAllTodos}>All</Label>
			  <Label id="toggle-all" onClick={this.toggleAll}>Toggle</Label>
			  <Label id="active-todo" onClick={this.props.displayActiveTodos}>Active</Label>
			  <Label id="complete-todo" onClick={this.props.displayCompletedTodos}>Completed</Label>
			</div>
		);

	}
}

Task.propTypes = {
  todos: PropTypes.array,
  displayAllTodos: PropTypes.func,
  toggleAll: PropTypes.func,
  displayActiveTodos: PropTypes.func,
  displayCompletedTodos: PropTypes.func,
  updateState: PropTypes.func
};

