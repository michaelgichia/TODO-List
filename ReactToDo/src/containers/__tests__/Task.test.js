import React from 'react';
import { mount, shallow } from 'enzyme';

import Task from '../Task';

describe('Task components', () => {
  const props = {};
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Task {...props}/>);
  });

  it('renders correctly', () => {
    const tree = shallow(<Task />);
    expect(tree).toMatchSnapshot();
  });

  it('should have a className attribute', () => {
    const renderedComponent = shallow(<Task />);
    expect(renderedComponent.prop('className')).toBeDefined();
  });
});
