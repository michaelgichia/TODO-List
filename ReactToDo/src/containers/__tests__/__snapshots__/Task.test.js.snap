// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task components renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="task"
>
    <Label
        id="all-todo"
        onClick={undefined}
    >
        All
    </Label>
    <Label
        id="toggle-all"
        onClick={[Function]}
    >
        Toggle
    </Label>
    <Label
        id="active-todo"
        onClick={undefined}
    >
        Active
    </Label>
    <Label
        id="complete-todo"
        onClick={undefined}
    >
        Completed
    </Label>
</div>,
  "nodes": Array [
    <div
      className="task"
>
      <Label
            id="all-todo"
            onClick={undefined}
      >
            All
      </Label>
      <Label
            id="toggle-all"
            onClick={[Function]}
      >
            Toggle
      </Label>
      <Label
            id="active-todo"
            onClick={undefined}
      >
            Active
      </Label>
      <Label
            id="complete-todo"
            onClick={undefined}
      >
            Completed
      </Label>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Task />,
      "_debugID": 12,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Task {
        "_reactInternalInstance": [Circular],
        "activeTodos": [Function],
        "completedTodos": [Function],
        "context": Object {},
        "newTodos": [Function],
        "props": Object {},
        "refs": Object {},
        "state": null,
        "toggleAll": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 8,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="task"
>
          <Label
                    id="all-todo"
                    onClick={undefined}
          >
                    All
          </Label>
          <Label
                    id="toggle-all"
                    onClick={[Function]}
          >
                    Toggle
          </Label>
          <Label
                    id="active-todo"
                    onClick={undefined}
          >
                    Active
          </Label>
          <Label
                    id="complete-todo"
                    onClick={undefined}
          >
                    Completed
          </Label>
</div>,
        "_debugID": 13,
        "_renderedOutput": <div
          className="task"
>
          <Label
                    id="all-todo"
                    onClick={undefined}
          >
                    All
          </Label>
          <Label
                    id="toggle-all"
                    onClick={[Function]}
          >
                    Toggle
          </Label>
          <Label
                    id="active-todo"
                    onClick={undefined}
          >
                    Active
          </Label>
          <Label
                    id="complete-todo"
                    onClick={undefined}
          >
                    Completed
          </Label>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Task />,
}
`;
